examples/features/broker-balancer/evenly-redirect/src/main/java/org/apache/activemq/artemis/jms/example/EvenlyRedirectExample.java

```diff
@@ -0,0 +1,102 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.jms.example;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Queue;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory;
+
+/**
+ * This example is demonstrating how incoming client connections are evely redirected from one broker towards
+ * worker brokers.
+ */
+public class EvenlyRedirectExample {
+
+   public static void main(final String[] args) throws Exception {
+
+      ConnectionFactory connectionFactoryClient0 = new ActiveMQConnectionFactory("tcp://localhost:61616?ha=true&reconnectAttempts=30&clientID=CLIENT_0");
+      ConnectionFactory connectionFactoryClient1 = new ActiveMQConnectionFactory("tcp://localhost:61616?ha=true&reconnectAttempts=30&clientID=CLIENT_1");
+
+      /**
+       * Step 1. Create a connection for producer0 and producer1, and send a few messages.
+       * the server0 will redirect the connection of each producer to a different target brokers.
+       */
+      Connection connectionProducer0 = null;
+      Connection connectionProducer1 = null;
+
+      try {
+         connectionProducer0 = connectionFactoryClient0.createConnection();
+         connectionProducer1 = connectionFactoryClient1.createConnection();
+
+         for (Connection connectionProducer : new Connection[] {connectionProducer0, connectionProducer1}) {
+            Session session = connectionProducer.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+            Queue queue = session.createQueue("exampleQueue");
+            MessageProducer sender = session.createProducer(queue);
+            for (int i = 0; i < 100; i++) {
+               sender.send(session.createTextMessage("Hello world n" + i + " - " + connectionProducer.getClientID()));
+            }
+         }
+      } finally {
+         if (connectionProducer0 != null) {
+            connectionProducer0.close();
+         }
+
+         if (connectionProducer1 != null) {
+            connectionProducer1.close();
+         }
+      }
+
+      /**
+       * Step 2. create a connection for consumer0 and consumer1, and receive a few messages.
+       * the server0 will redirect the connection to the same target broker of the respective producer
+       * because the consumer and the producer share the same clientID.
```

Comment:
```suggestion
       * the server0 will redirect the connection to the same target broker of the respective producer
       * from earlier as the new consumer connection uses the same ClientID.
```

Avoiding saying 'share the same ClientID' since only one connection at a time can use a ClientID

Reply:
Actually, as this uses 2 different ClientIDs but is doing 'least connections', is the ClientID relevant? When the second set of connections happens...what guarantee is it that they go to the same place as the producer if it is only doing 'least connections' balance and both servers have 0? Would it work as described if I changed the order of connection creation on the second round?

Reply:
The broker balancer defined int the evenly-redirect example uses a cache based on the CLIENT_ID key to provide affinity. I'll change the example to prove it.

Reply:
I missed to commit the updated example. Now it is updated and it uses `<target-key-filter>^.{3}</target-key-filter>`